### ISSUE TITLE
[DEMIS2SORMAS] Handle New Profile: Allow SORMAS to deal with DEMIS notifications containing multiple Specimens

### ISSUE THREAD

**Issue by FredrikSchaeferVitagroup** [hzi-braunschweig/SORMAS-Project#5598]
As this issue starts as a concept issue, it uses our new RFC template.

# Basic Info

- Feature Name: Allow SORMAS to deal with DEMIS notifications containing multiple Specimens
- Start Date: (today, 2021-06-01)
- [RFC PR](#rfc-pr): (once you open a PR, fill this in)
- [Tracking Issue](#tracking-issue): This issue

# Summary

_Give brief one or two paragraph explanation and summary of the feature or the changes you want to introduce._

The new [DEMIS profiles](https://simplifier.net/demis) allow several samples per DEMIS notification. This is fundamentally not compatible with the way SORMAS handles lab messages (which demis notifications are converted to in the demis-adapter), especially concerning the structure in the database and the processing workflow in the UI. The recommended solution is to restrict lab messages to one sample, and to create several lab messages out of one DEMIS notification with several samples.

# Motivation

_Start with the *Why*: Why do you propose this? List the most important user stories that support your use case and
highlight the user's goals and the desired outcome. Describe the status quo if necessary._

Status quo: The current DEMIS profiles support only one sample. Lab messages so far are designed to relate to only one sample. Lab messages are already used productively, users know the current work flow. By restricting lab messages to only one sample, the current work flow can be kept when dealing with DEMIS notifications holding several samples. This is the least confusing approach.

DEMIS notifications with several samples may be a rather rare thing. They might not even be used at all. The proposed approach is the simplest and therefore fits.

# High-Level Explanation and Requirements

_Put a checkmark on the *Why*, now we can continue with the *How*. General, high-level explanation and reasoning. Assume
the feature you propose is included in the system:
Explain how your users/stakeholders/developers should think about the feature and how they interact with it. What
impacts does your feature have on their way of working? Give concrete, real world examples.
Describe the status quo if necessary._

As stated above, lab message do not gain the ability to contain several samples (Specimens) and the adapter creates several lab message for one DEMIS notificaion with several Specimens.

## Requirements

_The capitalized keywords "MUST", "MUST NOT", "REQUIRED","SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
"RECOMMENDED"," MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119. You know the
drill :)_

Currently, the demis-adapter takes every part of a DEMIS notification, checks of which type it is (there are types like [Specimen](https://simplifier.net/covid-19labormeldung/specimensarscov2), [Observation](https://simplifier.net/covid-19labormeldung/pathogendetectionsarscov2)...) and handles them one after another. Handling means, it takes every relevant attribute of it and writes it's value to the related attribute in the lab message to be created for the DEMIS notification.
This is okay so far, as one DEMIS notification can have every relevant type only once. As stated above, this is going to change with the new profiles. When the same type occurred several times, the adapter would right now just overwrite the values previously written to the lab message. This MUST NOT  happen, as it induces a high risk of relevant information getting lost. Whenever the adapter handles an additional Specimen of a DEMIS notification, it must create a new lab message for it, containing all information relevant for that particular Specimen.

Each lab message SHOULD then be processed as usual (the current processing workflow will probably also be modified).

The adapter MUST NOT overwrite any values in a lab message, but this ticket is just about multiple Specimens.

# Detailed Design

_*What* do you propose exactly? Give a technical specification and details of your feature design. Also be specific about
how your feature relates and interacts with other systems._

# Drawbacks

_What are the disadvantages of your feature/idea/approach? What problems might be introduced with this? Are there
trade-offs or specific risk to your stakeholders? Can you come up with a reason why we should not do this? Remember,
there is no free lunch._

# Rationale and Alternatives

_Reason why the design you come up with is the best in the design space. Give insights into alternative designs that have
been explored but were discarded. Provide a brief decision log if appropriate. Are there (business) risks if we don't do
this? Are their potential equivalent designs that you have not explored yet?_

# Prior Art

_In case there is prior art, let us know. In case you tried something out, let us know
(the good, the bad, and the ugly). Where else has the feature been used and how did it go? Do you know what other people
or other communities tried and what experience they made? Can we learn a lesson from them? Further, if there are other
resources like research papers, articles, blog posts, or sketches please link them here for reference._

# Future Work

_Are there any long term goals or visions this RFC contributes to?_

# Open Questions

_Is there anything unclear? Is there anything not covered in this RFC that needs to be addressed separately?_
